Add support for SkyRukh Surge H7 board. (#31230)
ArduPilot/ardupilot: libraries/AP_HAL_ChibiOS/hwdef/SkyRukh_Surge_H7/README.md

# SkyRukh Surge H7 Flight Controller

![SkyRukh_Surge_H7.png](SkyRukh_Surge_H7.png)

The SkyRukh Surge H7 is a flight controller produced by SkyRukh.

## Features

 - MCU - STM32H743 32-bit processor running at 480 MHz
 - IMU - BMI088, ICM20602
 - Barometer - DPS310
 - OSD - MAX7456
 - microSD card slot
 - 7x UARTs
 - CAN port with JST-GH connector
 - External I2C port with JST-GH connector
 - 8x PWM Outputs (4 Motor Output with JST-GH connector)
 - Battery input voltage: 2S-6S
 - builtin RGB LED
 - BEC 3.3V 0.5A
 - BEC 5V 3A
 - BEC 9V 3A for VTX
 - Camera input with JST-GH connector

## Pinout

![SkyRukh Surge H7 Board Pinout Top](top_pinout.png "SkyRukh Surge H7 Pinout Top")
![SkyRukh Surge H7 Board Pinout Bottom](bottom_pinout.png "SkyRukh Surge H7 Pinout Bottom")

> ## UART Mapping

Reviewer: better table:
UART Port Mapping:
Port      UART      Protocol        TX DMA  RX DMA
--------- --------- ---------------- ------- -------
0         USB       MAVLink2        ✘       ✘
1         USART2    MAVLink2        ✔       ✔
2         USART6    MAVLink2        ✔       ✔
3         USART3    GPS             ✔       ✔
4         UART7     GPS             ✔       ✔
5         UART4     None            ✘       ✘
6         UART5     None            ✔       ✔
7         UART8     None            ✔       ✔

- SERIAL0 -> USB
- SERIAL1 -> USART2 (TELEM1)
- SERIAL2 -> USART6 (TELEM2)
- SERIAL3 -> USART3 (GPS1)
- SERIAL4 -> UART7  (None)
- SERIAL5 -> UART4  (Debug)
- SERIAL6 -> UART5  (None)
- SERIAL7 -> UART8  (None)

> The TELEM1 and TELEM2 ports have RTS/CTS pins, the other UARTs do not have RTS/CTS.

Reviewer: ```suggestion
The SERIAL1 and SERIAL2 ports have RTS/CTS pins, the other UARTs do not have RTS/CTS.
```

## RC Input

> RC input is provided through an SBUS connection with inverter control, offering direct SBUS receiver support.

Reviewer: ```suggestion
RC input is provided through the SBUS pad and supports all unidirectional RC protocols. For bi-directional protocols, such as CRSF/ELRS/SRXL2, a UART with DMA will be need to be used, See https://ardupilot.org/plane/docs/common-rc-systems.html for more information.
```

  
## OSD Support

The SkyRukh Surge H7 supports OSD using OSD_TYPE 1 (MAX7456 driver).

## PWM Output

The SkyRukh Surge H7 provides a total of 8 PWM outputs with DShot support.

PWM 1-4: Available on standard JST-SH motor connectors, compatible with typical ESC wiring.
PWM 5-8: Available on JST-GH connectors for additional motor/servo outputs.

All 8 outputs can be configured as PWM or DShot according to the user’s setup.

## Battery Monitoring

> The board has no built-in sensors but provides dedicated Battery Voltage Sensor and Battery Current Sensor ADC inputs, both accessible through the JST-SH ESC connector.

Reviewer: ```suggestion
The board has a internal voltage sensor and connections on the ESC connector for an external current sensor input. The voltage sensor can handle up to 6S LiPo batteries.

The default battery parameters are:

    - BATT_MONITOR = 4
    - BATT_VOLT_PIN = 16
    - BATT_CUR_PIN = 17 (C pin on ESC connector)
    - BATT_VOLT_MULT = 11.0 (assuming this is what the board has for its dvivder)
    - BATT_AMP_PERVLT = should be set according to ESC block used.

```
